deeplearning4j-core/src/test/java/org/deeplearning4j/nn/conf/constraints/TestConstraints.java

```diff
@@ -21,21 +22,401 @@
 
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
 public class TestConstraints {
 
     @Test
-    public void testConstraints() throws Exception {
+    public void testLayerCustomConstraints() throws Exception {
 
+        LayerConstraint[] constraints = new LayerConstraint[]{
+                new MaxNormConstraint(0.5, 1),
+                new MinMaxNormConstraint(0.3, 0.4, 1.0, 1),
+                new NonNegativeConstraint(),
+                new UnitNormConstraint(1)
+        };
+
+        Set<String> customConstraint = new HashSet<>();
+        customConstraint.add("RW");
```

Comment:
This customConstraint set isn't used anywhere?
We probably want' to check that the constraints are *not* applied to the other parameters too.